src/parser/esprima_cpp/esprima.cpp

```diff
@@ -4001,16 +3970,18 @@ class Parser {
             this->throwError(Messages::IllegalReturn);
         }
 
+        MetaNode node = this->createNode();
         this->expectKeyword(ReturnKeyword);
 
         bool hasArgument = !this->match(SemiColon) && !this->match(RightBrace) && !this->hasLineTerminator && this->lookahead.type != EOFToken;
         ASTNode argument = nullptr;
         if (hasArgument) {
+            node = this->createNode();
```

Comment:
Why this line is added?
IMO the firstly created node seems enough for location of return statement.